### PR TITLE
Niv/factory

### PR DESCRIPTION
**Pull request by nivcertora**
https://vaas-stg.certora.com/output/1512/9dc3d1eecb704343b6293bc08f5fb8fb?anonymousKey=8249f791cb2a32394c17f8e5a023f61ac945c64a

https://vaas-stg.certora.com/output/1512/2684c4e752e3443fbb8d3c04028eb722?anonymousKey=31889bde37843bc220db8d0a5b950221028d353c

https://vaas-stg.certora.com/output/1512/3b2fca5d4f384f538ada7661d6dcc9b0?anonymousKey=17ce8ce308cad255ea775f65f2d048ba7d49be31

https://vaas-stg.certora.com/output/1512/5a24f5501cd04675bb613c10045f5c0f?anonymousKey=2d51498c3baeb1d7cf702e78960cfb906fcf59a9

https://vaas-stg.certora.com/output/1512/450759d3262d4cacbeef57ccdf8ae761?anonymousKey=9ba731267ddb5d3ff0d471b31adac391084aa789

https://vaas-stg.certora.com/output/1512/9e233fc4f0274aef932232aa7251cf74?anonymousKey=eac38e439582d089b8ab6221a319aa0797358227

https://vaas-stg.certora.com/output/1512/1251e0ab70a4462899d2988619eaaa14?anonymousKey=ffc3b956b6c974ffe2de2a51361572678706989b

### CODE REVIEW

File: modules/passkey/contracts/SafeWebAuthnSignerFactory.sol [hunk 1]
```diff
@@ -36,6 +36,7 @@ contract SafeWebAuthnSignerFactory is ISafeSignerFactory {
         bytes32 codeHash = keccak256(
             abi.encodePacked(
                 type(SafeWebAuthnSignerProxy).creationCode,
+                "01234567891011121314152546", // Munging
```
Comment (yoav-el-certora) [inline modules/passkey/contracts/SafeWebAuthnSignerFactory.sol]: I think this should be in munging/harness folder, not on the real contract

File: certora/specs/SafeWebAuthnSignerFactory.spec [hunk 2]
```diff
@@ -87,4 +95,81 @@ rule createAndGetSignerEquivalence(){
...
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Correctness of Signer Creation. (Cant called twice and override) (Proved)                                           │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+ghost mathint numOfCreation;
+
+hook CREATE2(uint value, uint offset, uint length, bytes32 salt) address v{
+    require numOfCreation == numOfCreation + 1;
```
Comment (hristo-grigorov) [inline certora/specs/SafeWebAuthnSignerFactory.spec]: shouldn't it be 
`numOfCreation = numOfCreation + 1;`

File: certora/confs/SafeWebAuthnSignerFactory.conf [hunk 3]
```diff
@@ -1,16 +1,17 @@
 {
```
Comment (hristo-grigorov) [inline certora/confs/SafeWebAuthnSignerFactory.conf:1]: We should also add a sanity check:
`"rule_sanity": "basic",`

File: certora/specs/SafeWebAuthnSignerFactory.spec [hunk 4]
```diff
@@ -87,4 +95,81 @@ rule createAndGetSignerEquivalence(){
...
+    assert (a == proxy) => !hasNoCode(a);
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ isValidSignatureForSigner equiv to first deploying the signer with the factory, and then                            |
+|     verifying the signature with it directly (CERT-6221)                                                            │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule createAndVerifyEQtoIsValidSignatureForSigner()
```
Comment (hristo-grigorov) [inline certora/specs/SafeWebAuthnSignerFactory.spec:187]: this check is more like liveness

would be good to check also that both calls reverted (or did not) at the same time

File: certora/specs/SafeWebAuthnSignerFactory.spec (hunk 2 above)
Comment (nivcertora) [reply certora/specs/SafeWebAuthnSignerFactory.spec]: Your right : 
https://vaas-stg.certora.com/output/1512/fe6bf1f0ef354816bbfc0cc2343919c2?anonymousKey=ece8ef8072e3037c5f7b7e15277fc3ad2503aa39

File: certora/specs/SafeWebAuthnSignerFactory.spec (hunk 4 above)
Comment (nivcertora) [reply certora/specs/SafeWebAuthnSignerFactory.spec:187]: Right but for now we cant prove it

File: modules/passkey/contracts/SafeWebAuthnSignerFactory.sol (hunk 1 above)
Comment (nivcertora) [reply modules/passkey/contracts/SafeWebAuthnSignerFactory.sol]: Its not necessary it gets violated anyway

File: certora/confs/SafeWebAuthnSignerFactory.conf (hunk 3 above)
Comment (nivcertora) [reply certora/confs/SafeWebAuthnSignerFactory.conf:1]: Added thanks!

**Pull request merged**